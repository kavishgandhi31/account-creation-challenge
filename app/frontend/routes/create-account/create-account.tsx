import React from "react";
import Wealthfront_Logo from "../../reusable-components/wealthfront-logo/wealthfront-logo.png";
import { CreateAccountForm } from "../../reusable-components/create-account-form/create-account-form.tsx";

export function CreateAccount() {
    
    return (
        <div className="min-h-screen flex items-center justify-center bg-gray-25">
            <div className="flex flex-col items-center justify-center max-w-md w-full rounded-3xl shadow-2xl bg-white">
                <div className="w-full mt-4">
                    <img src={Wealthfront_Logo} alt="Wealthfront Logo" className="h-12 w-auto mx-auto mb-4" />
                </div>
                <h2 className="font-sans-serif font-black text-3xl text-gray-900">Create New Account</h2>
                <div className="flex flex-col items-center justify-center max-w-md w-full mt-8 px-4">
                    <CreateAccountForm />
                </div>
            </div>
        </div>
    )
}