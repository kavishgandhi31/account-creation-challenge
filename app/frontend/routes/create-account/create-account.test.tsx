import React from "react";
import '@testing-library/jest-dom';
import { render, screen, waitFor } from "@testing-library/react";
import { CreateAccount } from "./create-account";
import { BrowserRouter } from "react-router-dom";

describe("CreateAccount", () => {
    it ("renders the create account form", async () => {
        const { container } = render(<CreateAccount />, { wrapper: BrowserRouter });
        await waitFor(() => {
            expect(container).toMatchSnapshot();

            expect(screen.getByText("Create Account")).toBeInTheDocument();
            expect(screen.getByLabelText("Username")).toBeInTheDocument();
            expect(screen.getByLabelText("Password")).toBeInTheDocument();
            expect(screen.getByRole("button", { name: "Create Account" })).toBeInTheDocument();
        })
        
    });
})