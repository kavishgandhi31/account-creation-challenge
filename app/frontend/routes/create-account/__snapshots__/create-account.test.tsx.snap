// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount renders the create account form 1`] = `
<div>
  <div
    class="min-h-screen flex items-center justify-center bg-gray-25"
  >
    <div
      class="flex flex-col items-center justify-center max-w-md w-full rounded-3xl shadow-2xl bg-white"
    >
      <div
        class="w-full mt-4"
      >
        <img
          alt="Wealthfront Logo"
          class="h-12 w-auto mx-auto mb-4"
          src="test-file-stub"
        />
      </div>
      <h2
        class="font-sans-serif font-black text-3xl text-gray-900"
      >
        Create New Account
      </h2>
      <div
        class="flex flex-col items-center justify-center max-w-md w-full mt-8 px-4"
      >
        <form
          class="w-full"
        >
          <div>
            <div>
              <label
                class="block text-sm font-small text-gray-500 mb-4"
                for="Username"
              >
                Username
              </label>
              <input
                class="w-full px-3 border-b-2 border-gray-300 placeholder-gray-400 focus:outline-none focus:border-blue-500 sm:text-sm"
                id="Username"
                name="username"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="mt-4"
          >
            <div>
              <label
                class="block text-sm font-small text-gray-500 mb-4"
                for="Password"
              >
                Password
              </label>
              <input
                class="w-full px-3 border-b-2 border-gray-300 placeholder-gray-400 focus:outline-none focus:border-blue-500 sm:text-sm"
                id="Password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="pb-4 w-full items-center justify-center"
          >
            <button
              class="inline-block p-4 mt-4 w-full bg-[hsla(244,49%,80%,0.5)] text-white rounded-xl"
              disabled=""
              type="submit"
            >
              Create Account
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
